### PR TITLE
Implement user password recovering

### PR DESCRIPTION
**Pull request by anastasiia-bilova**
### Issue references
- Ticket (task) — https://github.com/Remmeauth/block-producers-directory-back/issues/28

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +111,53 @@ $ curl -X POST -d '{"old_password":"dmytro.striletskyi.1337", "new_password":"dm
 | old_password | Input arguments validation | This field is required.                            | 400         |
 | new_password | Input arguments validation | This field is required.                            | 400         |
 
+* `POST | /user/password/recovery` - password recovery for an existing user by his email.
```
Comment (dmytrostriletskyi) [inline README.md]: `Request password recovery for an existing user by his e-mail address. Send the recovery link to the e-mail address.`

File: README.md [hunk 2]
```diff
@@ -111,6 +111,53 @@ $ curl -X POST -d '{"old_password":"dmytro.striletskyi.1337", "new_password":"dm
...
+| Arguments | Type   | Required | Description  |
+| :-------: | :----: | :------: | ------------ |
+| email     | String | Yes      | User e-mail. |
+
+```bash
+$ curl -X POST -d '{"email":"dmytro.striletskyi@gmail.com"}' \
+      -H "Content-Type: application/json" \
+      http://localhost:8000/user/password/recovery | python -m json.tool
+{
+    "message": "Forgotten password has been sent to the specified e-mail address.",
```
Comment (dmytrostriletskyi) [inline README.md]: Outdated message. The message should say about recovery link.

File: directory/user/domain/objects.py [hunk 3]
```diff
@@ -53,3 +57,61 @@ def do(self, email, old_password, new_password):
...
+            raise UserWithSpecifiedEmailAddressDoesNotExistError
+
+        identifier = uuid.uuid4().hex
+
+        self.password_recovery_state.create(email=email, identifier=identifier)
+
+        return identifier
+
+
+class RecoveryUserPassword:
```
Comment (dmytrostriletskyi) [inline directory/user/domain/objects.py]: `Recovery` > `Recover`. Because we write verbs, not nouns.

File: directory/user/domain/objects.py [hunk 4]
```diff
@@ -53,3 +57,61 @@ def do(self, email, old_password, new_password):
...
+    """
+    Recovery user password implementation.
```
Comment (dmytrostriletskyi) [inline directory/user/domain/objects.py]: The same as above.

File: directory/user/domain/objects.py [hunk 5]
```diff
@@ -53,3 +57,61 @@ def do(self, email, old_password, new_password):
...
+    def __init__(self, user, password_recovery_state):
+        """
+        Constructor.
+        """
+        self.user = user
+        self.password_recovery_state = password_recovery_state
+
+    def do(self, user_identifier):
+        """
+        Recovery user password by user identifier.
```
Comment (dmytrostriletskyi) [inline directory/user/domain/objects.py]: The same as above.

File: directory/user/endpoints.py [hunk 6]
```diff
@@ -3,10 +3,16 @@
...
+    UserPasswordRecoverySingle,
+    UserPasswordSingle,
+    UserRequestPasswordRecoverySingle,
+)
 from user.views.registration import UserRegistrationSingle
 
 user_endpoints = [
     path('password/', UserPasswordSingle.as_view()),
+    path('password/recovery/', UserRequestPasswordRecoverySingle.as_view()),
+    path('password/recovery/<user_identifier>', UserPasswordRecoverySingle.as_view()),
```
Comment (dmytrostriletskyi) [inline directory/user/endpoints.py]: `password/recovery/<user_identifier>` > `password/recovery/<user_identifier>/`

File: README.md (hunk 1 above)
Comment (anastasiia-bilova) [reply README.md]: @dmytrostriletskyi, done.

File: README.md (hunk 2 above)
Comment (anastasiia-bilova) [reply README.md]: @dmytrostriletskyi, done.

File: directory/user/domain/objects.py (hunk 3 above)
Comment (anastasiia-bilova) [reply directory/user/domain/objects.py]: @dmytrostriletskyi, done.

File: directory/user/domain/objects.py (hunk 4 above)
Comment (anastasiia-bilova) [reply directory/user/domain/objects.py]: @dmytrostriletskyi, done.

File: directory/user/domain/objects.py (hunk 5 above)
Comment (anastasiia-bilova) [reply directory/user/domain/objects.py]: @dmytrostriletskyi, done.

File: directory/user/endpoints.py (hunk 6 above)
Comment (anastasiia-bilova) [reply directory/user/endpoints.py]: @dmytrostriletskyi, done.

**Pull request merged**